### PR TITLE
_create_tmp_config_dir() "mkdirs" the returned dir

### LINKED ISSUE #4120: _create_tmp_config_dir() in __init__.py does not create a directory
I believe the following code inserted after tempdir is created fixes it:
```
from matplotlib.cbook import mkdirs
mkdirs(tempdir)
```

### PR DESCRIPTION
**Pull request by curufinwe**
Fixes #4120

### CODE REVIEW

File: lib/matplotlib/__init__.py [hunk 1]
```diff
@@ -547,6 +548,11 @@ def _create_tmp_config_dir():
     tempdir = os.path.join(tempdir, 'matplotlib-%s' % getpass.getuser())
     os.environ['MPLCONFIGDIR'] = tempdir
 
+    try:
+        mkdirs(tempdir)
+    except OSError:
```
Comment (tacaswell) [inline lib/matplotlib/__init__.py]: I would not eat the exception here, this leaves the user in the same situation as before if this fails.

Comment (WeatherGod) [inline lib/matplotlib/__init__.py]: cbook.py already has a safe mkdirs(). Unfortunately, Python's mkdirs() has
some nasty race conditions and other problems, so you should use the safe
mkdirs.

**Comment by curufinwe** [comment]
I am using the cbook.py mkdirs.

**Comment by tacaswell** [comment]
Thanks for looking into this.  I am going to merge this as-is, but I think this needs a more-through overhaul to use `temfile.mkdtemp`  https://docs.python.org/2/library/tempfile.html#tempfile.mkdtemp which will do almost all of this work for us.

**Pull request merged**